packages/discord-request/src/storage-drivers/redis.ts

```diff
@@ -0,0 +1,51 @@
+import redis, { RedisClientType } from "redis";
+import type { StorageDriver } from "../types.js";
+
+interface RedisStorageDriverOptions {
+	url: string;
+	defaultTTL?: number;
+}
+
+export default class RedisStorageDriver implements StorageDriver {
+	client: RedisClientType;
+	defaultTTL = 300;
+	readyPromise: Promise<void> | undefined;
+
+	constructor({ url, defaultTTL }: RedisStorageDriverOptions) {
+		this.client = redis.createClient({
+			url,
+		});
+
+		// lock until ready
+		this.readyPromise = new Promise((resolve) => {
+			this.client.connect().then(resolve);
+		}).then(() => {
+			this.readyPromise = undefined;
+		});
+
```

Comment:
```suggestion
		this.ready = new Promise((resolve) => {
			return this.client.connect().then(resolve);
		})
```

I believe this will let you just write `await this.ready`